### ISSUE TITLE
Limit number of item to interact with

### ISSUE THREAD

**Issue by Salamafet** [rclone/rclone#4118]
Hi,

I try to delete the oldest file of a remote folder and copy the youngest file. The date of the files are never determined.
In the latest update (v1.51.0) we can order by modtime with the `--order-by modtime` (asc/desc) option.

Is it possible to add an option like `--limit` to interact with limited number of file ?
In my case: `--order-by modtime,asc --limit 1`

Thanks

**Comment by ncw** [comment]
> In the latest update (v1.51.0) we can order by modtime with the `--order-by modtime` (asc/desc) option.

These flags only work while transferring objects, not listing

I think adding a `--max-files` flag would probably work for "copying the youngest" file. Not sure about deleting the oldest though.

**Comment by Salamafet** [comment]
`--max-files` seems not to exist in the latest version of rclone and in the documentation 🤔

**Comment by ncw** [comment]
`--max-files` was only an idea for a flag!

**Comment by Salamafet** [comment]
Oh. Sorry my english is not so good.

Yes is a good idea to add `--max-files` flag.

**Comment by phillmac** [comment]
If I chucked a bounty on this of say $50 USD, is anybody interested?
And if so what's your preferred platform e.g. bountysource, issuehunt, gitpay?

**Comment by eharris** [comment]
This would also be extremely useful when trying to generate debug logs (using `-vv`) for filing bug reports.
I actually just went looking for this now, in order to try to limit a debug log for a bug report to a reasonable size and without a bunch of extraneous info.

I would also suggest that option this might be better as `--max-updates`, to make it clear that it only applies to files that are being copied/updated.  Or alternatively (even though it might be easily confused) `--max-transfers` (note the plural).  In that case, it might be good to change the existing `--max-transfer` to be `--max-transfer-size` to disambiguate.

**Comment by fjih** [comment]
eharris's comment about naming the flag `--max-updates` makes me think we need more clarification on the target use case of the feature.

On the surface, this feature seems akin to `--max-transfer`. Both define limits on the amount of transfer be it counting bytes or files. However, the difference lies in defining "transfer" vs "update". Transfer implies a movement of data from src to dst. Update is much more open to interpretation. Do deletes, renames, server-side copies, server-side moves, or modification time changes count as updates? Arguably they do.

Defining the target use case I think will help us clearly scope this feature.

I believe `--max-transefer` has a clear use case of limiting transfer size to hit quotas that are defined by size.
There isn't a correlated use case with file count limits given that the files can be of any size.

IMO, the original request seems better suited as a script rather than making rclone more monolithic. I would suggest the work there becomes adding support for `--order-by` to the `ls*` commands to make the sort easier for the user. From there they can call rclone directly on the file they decide to delete/copy.

That isn't to say that a feature for limiting the file transfers/updates is without value.

I do see a use is in limiting runaway modifications in a job. Other synchronization tools I've used have a concept of protecting the user against running a job larger than they expected. For instance, a job that should only copy 5 files and exit could instead fail to run as the tool detects that running the job would have resulted in 100 files copied. The user sets the parameters slightly above the expected size of the job, and the tool enforces the parameters to protect them should the expectation not be met. Depending on the use case, this could either be useful for limiting file transfers and/or updates (move, delete, rename, etc).
eharris's use case is also interesting, and I'm certain there are more uses that I haven't thought of.

Alternatively we could go the kitchen sink approach and try to cover everthing at the cost of complexity:
`--max-updates=N` and `--updates=string`
e.g. `--max-updates 10 --updates copy,move,delete`

TL;DR This feature likely needs more discussion to determine the scope of the work.

**Comment by BillTheHuman** [comment]
AI-assisted enquiry on William's behalf:

@phillmac, is your tentative $50 USD bounty offer from 2022 still of interest? If so, would a non-PayPal payment route be possible?

Before proposing work, we'd also want to resolve the scope raised above: should the limit count successful file transfers only, or deletions and metadata-only changes too? Concurrent workers and retries would need explicit acceptance checks.

This is a funding and scope enquiry, not a claim or a commitment to deliver the full feature for $50. No implementation or payment is being claimed.
